snapshotter/snapshotter.py

```diff
@@ -378,6 +394,13 @@ def snapshot(source, dest, debug=False, min_snapshots=3, extra_args=None):
     """
     date = _datetime()
     user, host, snapshots_root = _parse_path(dest)
+    if max_snapshots > -1 and max_snapshots - min_snapshots < 0:
+        raise InconsistentArgumentsError(
+            "--max-snapshots must be greater than --min-snapshots")
+    if max_snapshots > -1 and max_snapshots + 1 - min_snapshots >= 0:
+        snapshots = _ls_snapshots(dest)
+        for i in range(len(snapshots) - max_snapshots + 1):
+            _rm(snapshots[i], user, host, directory=True, debug=debug)
     while True:
```

Comment:
I think that maybe removing the old snapshots could be done more simply. Would this work?

```python
while len(_ls_snapshots(dest)) > max_snapshots - 1:
    _remove_oldest_snapshot(
        snapshots_root, user, host, min_snapshots=min_snapshots, debug=debug)
```